runtime/ejs-typedarrays.c

```diff
@@ -601,7 +738,291 @@ static ejsval                                                           \
+                                                                        \
+    /* 1. Let O be the result of calling ToObject passing the this value as the argument. */ \
+    ejsval O = ToObject(_this);                                         \
+                                                                        \
+    /* 3. Let lenVal be the result of Get(O, "length"). */              \
+    ejsval lenVal = Get(O, _ejs_atom_length);                           \
+                                                                        \
+    /* 4. Let len be ToLength(lenVal). */                               \
+    uint32_t len = ToLength(lenVal);                                    \
+                                                                        \
+    ejsval sep;                                                         \
+    const jschar *sep_str;                                              \
+    uint32_t sep_len;                                                   \
+                                                                        \
+    /* 6. If separator is undefined, let separator be the single-element String ",". */ \
+    /* 7. Let sep be ToString(separator). */                            \
+    if (EJSVAL_IS_UNDEFINED(separator)) {                               \
+        static jschar comma [] = {  (jschar)',' };                      \
+        sep_str = comma;                                                \
+        sep_len = 1;                                                    \
+    } else {                                                            \
+        sep = ToString(separator);                                      \
+        sep_str = EJSVAL_TO_FLAT_STRING(sep);                           \
+        sep_len = EJSVAL_TO_STRLEN(sep);                                \
+    }                                                                   \
+                                                                        \
+    /* 8. If len is zero, return the empty String. */                   \
+    if (len == 0)                                                       \
+        return _ejs_atom_empty;                                         \
+                                                                        \
+    /* (Prepare the internal buffer and pre calculate the length */     \
+    jschar *result;                                                     \
+    uint32_t result_len = 0;                                            \
+                                                                        \
+    ejsval *strings = (ejsval*)malloc (sizeof (ejsval) * len);          \
+    int i;                                                              \
+                                                                        \
+    for (i = 0; i < len; i++) {                                         \
+        strings[i] = ToString(Get(O, NUMBER_TO_EJSVAL(i)));             \
```

Comment:
this stores references to ejsvals in the heap without rooting them.

I just looked at `_ejs_Array_prototype_join` does the same thing.  filing an issue.